### ISSUE TITLE
[ngalert] Missing alert notifications in Grafana 8.1.3 and 8.1.4

### ISSUE THREAD

**Issue by gerobinson** [grafana/grafana#39295]
**What happened**:

Some users are reporting missing alert notifications on Grafana 8.1.3 and 8.1.4 which is fixed when downgrading to 8.1.2.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.3/8.1.4
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by CirnoT** [comment]
Confirmed, no email alerts since v8.1.3

```
[smtp]
enabled = true
host = 127.0.0.1:25
from_address = ***
from_name = ***

[alerting]
enabled = true
execute_alerts = true
error_or_timeout = alerting
nodata_or_nullvalues = no_data
concurrent_render_limit = 5
max_attempts = 3

[feature_toggles]
enable = ngalert
```

```
{
  "template_files": {},
  "alertmanager_config": {
    "route": {
      "receiver": "***"
    },
    "templates": null,
    "receivers": [
      {
        "name": "***",
        "grafana_managed_receiver_configs": [
          {
            "uid": "bB1fvweGk",
            "name": "***",
            "type": "email",
            "disableResolveMessage": false,
            "settings": {
              "addresses": "***",
              "singleEmail": false
            },
            "secureFields": {}
          }
        ]
      }
    ]
  }
}
```

**Comment by torkelo** [comment]
so 8.1.4 did not fix this? https://github.com/grafana/grafana/pull/38983

**Comment by CirnoT** [comment]
Considering that I have created this configuration for ngalert _after_ v7.2 and that encryption does not cover email receivers, I believe that this would have absolutely nothing to do with reported issue.

Others seem to confirm that the issue occurs on 8.1.4 as well (check https://github.com/grafana/grafana/issues/39009#issuecomment-920812212)

I also don't see how linked PR could affect only 8.1.3+ but not 8.1.2.

---

On unrelated note, the linked PR fixed migration however for affected users this will change nothing, as migration already ran.

**Comment by chaoyi996** [comment]
Not notified on 8.1.3 and 8.1.4

**Comment by haeiven** [comment]
Having the same issue with 8.2.0-beta1.

**Comment by ImmoWetzel** [comment]
no notification also for grafana 8.1.4 as docker container

**Comment by staeglis** [comment]
I can confirm the issue too

**Comment by ImmoWetzel** [comment]
I do see 
t=2021-09-20T15:30:52+0000 lvl=dbug msg="Applying default URL parsing for this data source type" logger=datasource type=prometheus url=http://10.160.7.44:9090
t=2021-09-20T15:30:52+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/api/ruler/1/api/v1/rules **_status=500_** remote_addr=172.20.62.163 time_ms=2 size=96 referer=http://10.160.7.44:3000/alerting/list

in my logs... Does this helps

**Comment by jan666** [comment]
Same here. E-Mail + Pushover does not work (nothing else configures)

**Comment by gerobinson** [comment]
I'm afraid I have not been able to reproduce this issue on either 8.1.3 or 8.1.4. It doesn't appear to be related to email notifications as others have replied this affects other notifications too, such as Pushover.

This is the configuration I have been using:

```
{
  "template_files": {},
  "alertmanager_config": {
    "route": {
      "receiver": "test-email"
    },
    "templates": null,
    "receivers": [
      {
        "name": "test-email",
        "grafana_managed_receiver_configs": [
          {
            "uid": "2c_9eVHnzz",
            "name": "test-email",
            "type": "email",
            "disableResolveMessage": false,
            "settings": {
              "addresses": "example@example.com",
              "singleEmail": false
            },
            "secureFields": {}
          }
        ]
      }
    ]
  }
}
```

and I created an alert rule with the following configuration:

![Screenshot 2021-09-21 at 14 21 24](https://user-images.githubusercontent.com/85952834/134178211-5eb92d5f-c578-4147-953e-d47a5f4dd60d.png)

```
{
  "status": "success",
  "data": {
    "groups": [
      {
        "name": "George test",
        "file": "gdev dashboards",
        "rules": [
          {
            "state": "firing",
            "name": "George test",
            "query": "[{\"refId\":\"A\",\"queryType\":\"\",\"relativeTimeRange\":{\"from\":21600,\"to\":0},\"datasourceUid\":\"WR_h6VH7z\",\"model\":{\"exemplar\":true,\"expr\":\"go_goroutines\",\"interval\":\"\",\"intervalMs\":1000,\"legendFormat\":\"\",\"maxDataPoints\":43200,\"refId\":\"A\"}},{\"refId\":\"B\",\"queryType\":\"\",\"relativeTimeRange\":{\"from\":0,\"to\":0},\"datasourceUid\":\"-100\",\"model\":{\"conditions\":[{\"evaluator\":{\"params\":[3],\"type\":\"gt\"},\"operator\":{\"type\":\"and\"},\"query\":{\"params\":[\"A\"]},\"reducer\":{\"params\":[],\"type\":\"last\"},\"type\":\"query\"}],\"datasource\":\"__expr__\",\"expression\":\"A\",\"hide\":false,\"intervalMs\":1000,\"maxDataPoints\":43200,\"reducer\":\"max\",\"refId\":\"B\",\"type\":\"reduce\"}},{\"refId\":\"C\",\"queryType\":\"\",\"relativeTimeRange\":{\"from\":0,\"to\":0},\"datasourceUid\":\"-100\",\"model\":{\"conditions\":[{\"evaluator\":{\"params\":[0,0],\"type\":\"gt\"},\"operator\":{\"type\":\"and\"},\"query\":{\"params\":[]},\"reducer\":{\"params\":[],\"type\":\"avg\"},\"type\":\"query\"}],\"datasource\":\"__expr__\",\"expression\":\"$B \\u003e 0\",\"hide\":false,\"intervalMs\":1000,\"maxDataPoints\":43200,\"refId\":\"C\",\"type\":\"math\"}}]",
            "duration": 30,
            "annotations": {
              "__dashboardUid__": "9HYAe4H7k",
              "__panelId__": "2"
            },
            "alerts": [
              {
                "labels": {
                  "__alert_rule_namespace_uid__": "ZnHh6VNnz",
                  "__alert_rule_uid__": "6_xJ6VHnz",
                  "__name__": "go_goroutines",
                  "alertname": "George test",
                  "instance": "localhost:9090",
                  "job": "prometheus"
                },
                "annotations": {
                  "__dashboardUid__": "9HYAe4H7k",
                  "__panelId__": "2"
                },
                "state": "Alerting",
                "activeAt": "2021-09-21T14:18:45.994011+01:00",
                "value": "[ var='C' labels={__name__=go_goroutines, instance=localhost:9090, job=prometheus} value=1 ], [ var='B' labels={__name__=go_goroutines, instance=localhost:9090, job=prometheus} value=36 ]"
              }
            ],
            "labels": {},
            "health": "ok",
            "lastError": "",
            "type": "alerting",
            "lastEvaluation": "2021-09-21T14:19:45.994011+01:00",
            "evaluationTime": 0.014152834
          }
        ],
        "interval": 10,
        "lastEvaluation": "2021-09-21T14:19:45.994011+01:00",
        "evaluationTime": 0
      }
    ]
  }
}
```

I can see that Grafana is attempting to send an email with both smtp to localhost (which I do not have a listening smtp server) and I also confirmed that a webhook notification can be sent to an HTTP server that I have also listening on localhost.

```
level=error component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="test-email/email[0]: notify retry canceled due to unrecoverable error after 1 attempts: Failed to send notification to email addresses: example@example.com: dial tcp [::1]:25: connect: connection refused"
```

**Comment by gerobinson** [comment]
I'm afraid I have not been able to reproduce this issue on either 8.1.3 or 8.1.4. It doesn't appear to be related to email notifications as others have replied this affects other notifications too, such as Pushover.

This is the configuration I have been using:

```
{
  "template_files": {},
  "alertmanager_config": {
    "route": {
      "receiver": "test-email"
    },
    "templates": null,
    "receivers": [
      {
        "name": "test-email",
        "grafana_managed_receiver_configs": [
          {
            "uid": "2c_9eVHnzz",
            "name": "test-email",
            "type": "email",
            "disableResolveMessage": false,
            "settings": {
              "addresses": "example@example.com",
              "singleEmail": false
            },
            "secureFields": {}
          }
        ]
      }
    ]
  }
}
```

and I created an alert rule with the following configuration:

![Screenshot 2021-09-21 at 14 28 24](https://user-images.githubusercontent.com/85952834/134179384-593a4e17-4234-427a-8210-f370bc3ba0d2.png)

I can see that Grafana is attempting to send an email with both smtp to localhost (which I do not have a listening smtp server) and I also confirmed that a webhook notification can be sent to an HTTP server that I have also listening on localhost.

```
level=error component=dispatcher msg="Notify for alerts failed" num_alerts=1 err="test-email/email[0]: notify retry canceled due to unrecoverable error after 1 attempts: Failed to send notification to email addresses: example@example.com: dial tcp [::1]:25: connect: connection refused"
```

**Comment by staeglis** [comment]
@gerobinson I assume the issue affects alerts directly managed by grafana only.

**Comment by ImmoWetzel** [comment]
I use ngalert with prometheus as datasource

***@***.***

But I also used a contact point which contains email and MS Teams…
Maybe just tell me how to create required logs

With kind regards

Immo Wetzel
Office: +49 3834 5352 823 ; Cell: +49 151 1472 9225 ; Email: ***@***.******@***.***> ;
This message has been classified General Business by Immo Wetzel on Tuesday, September 21, 2021 at 4:57:27 PM.

**Comment by DasSkelett** [comment]
I am also suffering from missing notifications for Grafana managed alerts, currently on Grafana v8.1.4.
In my case those notifications should happen via Telegram.

```
t=2021-09-21T18:03:14+0000 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-21T18:03:14+0000

t=2021-09-21T18:03:17+0000 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=\"Request Duration (logarithmic)[b74ae7b][resolved]\"" logger=alertmanager

t=2021-09-21T18:03:17+0000 lvl=info msg="level=debug component=dispatcher aggrGroup={}:{} msg=flushing alerts=\"[Request Duration (logarithmic)[b74ae7b][resolved]]\"" logger=alertmanager

t=2021-09-21T18:06:14+0000 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-21T18:06:14+0000

t=2021-09-21T18:06:17+0000 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=\"Request Duration (logarithmic)[e4c5e85][resolved]\"" logger=alertmanager
```

What confuses me is that they all contain this `[resolved]` tag. I don't know what it means, but the alert is definitely not resolved, and the UI correctly shows it as "Firing":
![UI](https://user-images.githubusercontent.com/28812678/134225546-7a9e5ecb-b99b-4215-9d90-617c23bf0ac4.png)


Happy to provide other details and logs about my instance.

**Comment by vfylyk** [comment]
On version 8.1.4, I have recreated more or less what George did. It still did not issue any notifications, and in their first appearance in the logs, everything displays as "resolved".

I created this contact point:

<img width="1239" alt="image" src="https://user-images.githubusercontent.com/48038828/134271966-d34a341d-b2a0-4028-8a8d-ad53c6ae719f.png">

Then this notification policy:

<img width="1300" alt="image" src="https://user-images.githubusercontent.com/48038828/134272033-ce0d3c69-836e-4ece-8ab2-125c9dc20cd6.png">

And the alert with these settings:

<img width="1294" alt="image" src="https://user-images.githubusercontent.com/48038828/134272773-0d0a647d-c90f-409f-a4c6-417f1ec33118.png">
<img width="1280" alt="image" src="https://user-images.githubusercontent.com/48038828/134272802-14e26f3e-6398-42eb-adf1-1f3f5085dc03.png">
<img width="1299" alt="image" src="https://user-images.githubusercontent.com/48038828/134272876-e2c7acc6-99ef-4a3c-85e2-4709d9b12c56.png">

This is what comes up in the logs as the first mention of these test alerts. No notification is ever attempted to be sent.

```
t=2021-09-22T02:21:18+0000 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=Test[aee0740][resolved]" logger=alertmanager
t=2021-09-22T02:21:18+0000 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=Test[3063b92][resolved]" logger=alertmanager
t=2021-09-22T02:21:29+0000 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T02:21:29+0000
t=2021-09-22T02:21:44+0000 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T02:21:44+0000
t=2021-09-22T02:21:48+0000 lvl=info msg="level=debug component=dispatcher aggrGroup=\"{}/{alertkey=\\\"test\\\"}:{}\" msg=flushing alerts=\"[Test[aee0740][resolved] Test[3063b92][resolved]]\"" logger=alertmanager
t=2021-09-22T02:21:59+0000 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T02:21:59+0000
t=2021-09-22T02:22:14+0000 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T02:22:14+0000
t=
```

**Comment by jan666** [comment]
I created a simple Testalert:

Alert name = Test
Alert type = Grafana managed Alert
Folder = General Alerting

2 Contact points (Pushover and E-Mail)  - worked before 8.1.3

Debug output:

```
t=2021-09-22T07:25:05+0200 lvl=dbug msg="Influxdb request" logger=tsdb.influxdb url="[REDACTED]"
t=2021-09-22T07:25:05+0200 lvl=dbug msg="expression datasource query (seriesSet)" logger=expr query=A
t=2021-09-22T07:25:05+0200 lvl=dbug msg="state manager processing evaluation results" logger=ngalert uid=BlTgAA47k resultCount=1
t=2021-09-22T07:25:05+0200 lvl=dbug msg="setting alert state" logger=ngalert uid=BlTgAA47k
t=2021-09-22T07:25:05+0200 lvl=dbug msg="saving alert states" logger=ngalert count=1
t=2021-09-22T07:25:06+0200 lvl=dbug msg="sending alerts to notifier" logger=ngalert count=0 alerts=[]
t=2021-09-22T07:25:08+0200 lvl=dbug msg="new alert rule version fetched" logger=ngalert title=Test key="{orgID: 1, UID: -kkTyHHnk}" version=2
t=2021-09-22T07:25:08+0200 lvl=dbug msg="Received a query request" logger=tsdb.influxdb numQueries=1
t=2021-09-22T07:25:08+0200 lvl=dbug msg="Making a non-Flux type query" logger=tsdb.influxdb
t=2021-09-22T07:25:08+0200 lvl=dbug msg="Influxdb request" logger=tsdb.influxdb url="[REDACTED]"
t=2021-09-22T07:25:08+0200 lvl=dbug msg="expression datasource query (seriesSet)" logger=expr query=A
t=2021-09-22T07:25:08+0200 lvl=dbug msg="state manager processing evaluation results" logger=ngalert uid=-kkTyHHnk resultCount=1
t=2021-09-22T07:25:08+0200 lvl=dbug msg="setting alert state" logger=ngalert uid=-kkTyHHnk
t=2021-09-22T07:25:08+0200 lvl=dbug msg="saving alert states" logger=ngalert count=1
t=2021-09-22T07:25:08+0200 lvl=dbug msg="alert state changed creating annotation" logger=ngalert alertRuleUID=-kkTyHHnk newState=Alerting
t=2021-09-22T07:25:08+0200 lvl=dbug msg="sending alerts to notifier" logger=ngalert count=1 alerts="[{Annotations:map[__value_string__:[ metric='messung.Sensor1' labels={} value=21.5 ]] EndsAt:2021-09-22T07:25:00.782+02:00 StartsAt:2021-09-22T07:25:00.782+02:00 Alert:{GeneratorURL:[REDACTED] Labels:map[__alert_rule_namespace_uid__:vs37g6g7k __alert_rule_uid__:-kkTyHHnk alertname:Test]}}]"
t=2021-09-22T07:25:08+0200 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=Test[213cdf7][resolved]" logger=alertmanager
t=2021-09-22T07:25:10+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
```

**Comment by staeglis** [comment]
> I created a simple Testalert:
> 
> Alert name = Test
> Alert type = Grafana managed Alert
> Folder = General Alerting
> 
> 2 Contact points (Pushover and E-Mail) - worked before 8.1.3
> 
> Debug output:
> 
> ```
> t=2021-09-22T07:25:05+0200 lvl=dbug msg="Influxdb request" logger=tsdb.influxdb url="[REDACTED]"
> t=2021-09-22T07:25:05+0200 lvl=dbug msg="expression datasource query (seriesSet)" logger=expr query=A
> t=2021-09-22T07:25:05+0200 lvl=dbug msg="state manager processing evaluation results" logger=ngalert uid=BlTgAA47k resultCount=1
> t=2021-09-22T07:25:05+0200 lvl=dbug msg="setting alert state" logger=ngalert uid=BlTgAA47k
> t=2021-09-22T07:25:05+0200 lvl=dbug msg="saving alert states" logger=ngalert count=1
> t=2021-09-22T07:25:06+0200 lvl=dbug msg="sending alerts to notifier" logger=ngalert count=0 alerts=[]
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="new alert rule version fetched" logger=ngalert title=Test key="{orgID: 1, UID: -kkTyHHnk}" version=2
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="Received a query request" logger=tsdb.influxdb numQueries=1
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="Making a non-Flux type query" logger=tsdb.influxdb
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="Influxdb request" logger=tsdb.influxdb url="[REDACTED]"
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="expression datasource query (seriesSet)" logger=expr query=A
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="state manager processing evaluation results" logger=ngalert uid=-kkTyHHnk resultCount=1
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="setting alert state" logger=ngalert uid=-kkTyHHnk
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="saving alert states" logger=ngalert count=1
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="alert state changed creating annotation" logger=ngalert alertRuleUID=-kkTyHHnk newState=Alerting
> t=2021-09-22T07:25:08+0200 lvl=dbug msg="sending alerts to notifier" logger=ngalert count=1 alerts="[{Annotations:map[__value_string__:[ metric='messung.Sensor1' labels={} value=21.5 ]] EndsAt:2021-09-22T07:25:00.782+02:00 StartsAt:2021-09-22T07:25:00.782+02:00 Alert:{GeneratorURL:[REDACTED] Labels:map[__alert_rule_namespace_uid__:vs37g6g7k __alert_rule_uid__:-kkTyHHnk alertname:Test]}}]"
> t=2021-09-22T07:25:08+0200 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=Test[213cdf7][resolved]" logger=alertmanager
> t=2021-09-22T07:25:10+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
> ```
> 
> I just downgraded to 8.1.2 and changed nothing else. After restarting I immediately received the Pushover notification:
> 
> ```
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="new alert rule version fetched" logger=ngalert title=Test key="{orgID: 1, UID: -kkTyHHnk}" version=5
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="Received a query request" logger=tsdb.influxdb numQueries=1
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="Making a non-Flux type query" logger=tsdb.influxdb
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="Influxdb request" logger=tsdb.influxdb url="[REDACTED]"
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="expression datasource query (seriesSet)" logger=expr query=A
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="state manager processing evaluation results" logger=ngalert uid=-kkTyHHnk resultCount=1
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="setting alert state" logger=ngalert uid=-kkTyHHnk
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="saving alert states" logger=ngalert count=1
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="alert state changed creating annotation" logger=ngalert alertRuleUID=-kkTyHHnk newState=Alerting
> t=2021-09-22T07:43:17+0200 lvl=dbug msg="sending alerts to notifier" logger=ngalert count=1 alerts="[{Annotations:map[__value_string__:[ metric='messung.Sensor1' labels={} value=21.7 ]] EndsAt:2021-09-22T07:45:09.528+02:00 StartsAt:2021-09-22T07:43:09.528+02:00 Alert:{GeneratorURL:[REDACTED] Labels:map[__alert_rule_namespace_uid__:vs37g6g7k __alert_rule_uid__:-kkTyHHnk alertname:Test]}}]"
> t=2021-09-22T07:43:17+0200 lvl=info msg="level=debug component=dispatcher msg=\"Received alert\" alert=Test[213cdf7][active]" logger=alertmanager
> t=2021-09-22T07:43:19+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
> t=2021-09-22T07:43:24+0200 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T07:43:24+0200
> t=2021-09-22T07:43:29+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
> t=2021-09-22T07:43:39+0200 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T07:43:39+0200
> t=2021-09-22T07:43:39+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
> t=2021-09-22T07:43:39+0200 lvl=dbug msg="neither config nor template have changed, skipping configuration sync." logger=alertmanager
> t=2021-09-22T07:43:47+0200 lvl=info msg="level=debug component=dispatcher aggrGroup={}:{} msg=flushing alerts=[Test[213cdf7][active]]" logger=alertmanager
> t=2021-09-22T07:43:47+0200 lvl=dbug msg="Sending webhook" logger=notifications url=https://api.pushover.net/1/messages.json http method=POST
> t=2021-09-22T07:43:48+0200 lvl=dbug msg="Webhook succeeded" logger=notifications url=https://api.pushover.net/1/messages.json statuscode="200 OK"
> t=2021-09-22T07:43:48+0200 lvl=info msg="level=debug component=dispatcher receiver=pushover-jan integration=pushover[0] msg=\"Notify success\" attempts=1" logger=alertmanager
> t=2021-09-22T07:43:49+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
> t=2021-09-22T07:43:54+0200 lvl=info msg="recording state cache metrics" logger=ngalert now=2021-09-22T07:43:54+0200
> t=2021-09-22T07:43:59+0200 lvl=dbug msg="alert rules fetched" logger=ngalert count=4
> ```
> 
> Maybe a side note, but: the Tab "Notifications" in 8.1.3 was empty. In 8.1.2 there is a list of groups. "No grouping" has the firing alert.

I've observed this too

**Comment by gerobinson** [comment]
I have found the issue and identified the change in 8.1.3 that caused it. The bug occurs when an alert rule is configured to evaluate at an interval that is greater than the interval at which Grafana resends firing alerts to the internal Alertmanager to stop them resolving, which in the 8.1.3 and 8.1.4 releases is 30 seconds. To fix notifications in 8.1.3 and 8.1.4 it is acceptable to set the alert evaluation interval for an alert rule to an interval that is less than 30 seconds.

**Comment by gerobinson** [comment]
I have found the issue and identified the change in 8.1.3 that caused it. The bug occurs when an alert rule is configured to evaluate at an interval that is greater than the interval at which Grafana resends firing alerts to the internal Alertmanager to stop them resolving, which in the 8.1.3 and 8.1.4 releases is 30 seconds. To fix notifications in 8.1.3 and 8.1.4 it is acceptable to set the alert evaluation interval for an alert rule to an interval that is 30 seconds or lower.

**Comment by jan666** [comment]
Can confirm. Setting "Evaluate every" to "20s" (instead of 1m) and it works again (8.1.3).

Thanks!
